app/src/server/trpc/routers/draft.ts

```diff
@@ -36,5 +43,11 @@ export const draftRouter = router({
     .input(z.object({ resourceType: z.enum(['Measure', 'Library']), draft: z.any(), id: z.string() }))
     .mutation(async ({ input }) => {
       return updateDraft(input.resourceType, input.id, input.draft);
+    }),
+
+  deleteDraft: publicProcedure
+    .input(z.object({ id: z.string().optional(), resourceType: z.enum(['Measure', 'Library']).optional() }))
```

Comment:
I'm still getting up to speed on trpc, but is there a reason why we set both properties (id and resourceType) to optional, but then immediately check they exist before deleting? Should we just not mark them as optional?

Reply:
After doing some more testing here, it doesn't seem like we need to make the inputs optional as I originally did here.
In general, some of the draft procedures throw TRPC client errors when the page first loads because the FHIR resource type will be undefined on first load. To avoid this TRPC error, procedures like `getDrafts` and `getDraftById` have optional inputs that get checked before executing the query. Since `deleteDraft` does not get executed on first load (it's a mutation that gets triggered by a button click), I am not getting any TRPC client errors when the inputs are required instead of optional. I'll update this!